### PR TITLE
More usage collection in parallel, less webhooks in parallel

### LINKED ISSUE #1765: Rate limiting and parallelism limits for webhook delivery
We just had an incident where a misconfigured webhook (pointed at a service that no longer existed) trivially takes down a Singularity in a way that neither crashes nor is recoverable.  The file descriptor table (left at a small-ish default 4096 entries) filled entirely with webhook dispatches, and the rate that they were created exceeded the rate that they failed.   This led to many IOExceptions and SocketExceptions saying, 'out of file descriptors'.

The same webhook should have a limited number of parallel requests to it, and possibly some kind of circuit breaker to protect against failed webhooks.

Different webhooks should be reasonably separated so that one failure cannot affect others

The total number of webhook dispatches should be limited so an administrator can ensure it does not bring down the rest of Singularity

### PR DESCRIPTION
**Pull request by ssalinas**
/fixes #1765

### CODE REVIEW

File: SingularityService/src/main/java/com/hubspot/singularity/hooks/SingularityWebhookSender.java [hunk 1]
```diff
@@ -60,23 +74,30 @@ public void checkWebhooks() {
     int requestUpdates = 0;
     int deployUpdates = 0;
 
-    for (SingularityWebhook webhook : webhooks) {
+    List<CompletableFuture<Void>> webhookFutures = new ArrayList<>();
```
Comment (baconmania) [inline SingularityService/src/main/java/com/hubspot/singularity/hooks/SingularityWebhookSender.java]: Is this list ever written to?

File: SingularityService/src/main/java/com/hubspot/singularity/hooks/SingularityWebhookSender.java [hunk 2]
```diff
@@ -39,13 +47,19 @@
...
   @Inject
   public SingularityWebhookSender(SingularityConfiguration configuration, AsyncHttpClient http, ObjectMapper objectMapper, TaskHistoryHelper taskHistoryHelper, WebhookManager webhookManager) {
     this.configuration = configuration;
     this.http = http;
     this.webhookManager = webhookManager;
     this.taskHistoryHelper = taskHistoryHelper;
     this.objectMapper = objectMapper;
+
+    this.webhookSemaphore = AsyncSemaphore.newBuilder(configuration::getMaxConcurrentWebhooks).build();
+    this.webhookExecutorService = Executors.newCachedThreadPool(new ThreadFactoryBuilder().setNameFormat("webhooks-%d").build());
```
Comment (baconmania) [inline SingularityService/src/main/java/com/hubspot/singularity/hooks/SingularityWebhookSender.java]: Doesn't look like this guy is used anywhere after being created.

**Comment by baconmania** [comment]
🚢

**Comment by pschoenfelder** [comment]
🚢

**Comment by stevenschlansker** [comment]
Thanks!

**Pull request merged**